Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2008,36 +2010,222 @@ namespace OpenInfraPlatform {
+				{
+
+					return placementConverter->convertIfcCartesianPoint(cartesianPoint);
+				}
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a trimming point on the curve using \c IfcParameterValue.
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] parameter				A pointer to data form \c IfcParameterValue.
+				* \return							The location of the trimming point.
+				*/
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
+					const typename IfcEntityTypesT::IfcParameterValue & parameter) const throw(...)
+				{
+					double angle = parameter * UnitConvert()->getAngleInRadianFactor();
+					// determine position
+					carve::math::Matrix conicPositionMatrix = placementConverter->convertIfcAxis2Placement(circle->Position);
+
+					// Get radius
+					double circleRadius = circle->Radius * UnitConvert()->getLengthInMeterFactor();
+
+					carve::geom::vector<3> circleCenter =
+						conicPositionMatrix * carve::geom::VECTOR(0, 0, 0);
+					return circleCenter + carve::geom::VECTOR(circleRadius * cos(angle), circleRadius * sin(angle), 0.);
+				}
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a trimming point on the Curve.
+				* \param[in] ellipse				A pointer to data from \c IfcEllipse.
+				* \param[in] trimming				A pointer to data form \c IfcTrimmingSelect.
+				* \return							The location of the trimming point.
+				*/
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcEllipse>& ellipse,
+					const typename IfcEntityTypesT::IfcTrimmingSelect & trimming) const throw(...)
+				{
+					switch (trimming.which())
+					{
+					case 0:
+					{
+						// Calculate a trimming point using \c IfcCartesianPoint. 
+						return getPointOnCurve(ellipse, trimming.get<0>());
+					}
+					case 1:
+					{
+						// Calculate a trimming point using \c IfcParameterValue.
+						return getPointOnCurve(ellipse, trimming.get<1>());
+					}
+					default:
+						throw oip::InconsistentGeometryException(ellipse, "IfcTrimmingSelect is wrong!");
+					}
+				}
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a trimming point on the curve using \c IfcCartesianPoint. 
+				* \param[in] ellipse				A pointer to data from \c IfcEllipse.
+				* \param[in] cartesianPoint			A pointer to data form \c IfcCartesianPoint.
+				* \return							The location of the trimming point.
+				*/
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcEllipse>& ellipse,
+					const EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>& cartesianPoint) const throw(...)
+				{
+
+					return placementConverter->convertIfcCartesianPoint(cartesianPoint);
+				}

```

Comment:
This looks very very similar to `getPointOnCurve` for `IfcCircle` - consider using templates / base classes.

Reply:
Resolved: ad49034